src/update-feeds-loops/submit-transactions.ts

```diff
@@ -31,8 +31,8 @@ export const createUpdateFeedCalldatas = (api3ServerV1: Api3ServerV1, updatableD
     ])
   );
 
-  // If there are multiple beacons in the data feed it's a beacons set which we need to update as well.
-  return allBeacons.length > 1
+  // If there are multiple beacons in the data feed it's a beacons set which we may need to update as well.
+  return allBeacons.length > 1 && shouldUpdateBeaconSet
```

Comment:
Maybe we can rely only on `shouldUpdateBeaconSet`. By reading the code the other check looks extranous.

Reply:
https://github.com/api3dao/airseeker/pull/343/commits/831224d7c3131f9ef443db7139e98328774e4144